src/main/scala/com/johnsnowlabs/nlp/datasets/POS.scala

```diff
@@ -0,0 +1,114 @@
+  }
+
+  /*
+   * This section is to help users to convert text files in token|tag style into DataFrame
+   * with POS Annotation for training PerceptronApproach
+   * */
+
+  private def annotateTokensTags: UserDefinedFunction = udf { (tokens: Seq[String], tags: Seq[String], text: String) =>
+    lazy val strTokens = tokens.mkString("#")
+    lazy val strPosTags = tags.mkString("#")
+
+    require(tokens.length == tags.length, s"Cannot train from DataFrame since there" +
+      s" is a row with different amount of tags and tokens:\n$strTokens\n$strPosTags")
+
+    val tokenTagAnnotation: ArrayBuffer[Annotation] = ArrayBuffer()
+    def annotatorType: String = AnnotatorType.POS
+    var lastIndex = 0
+
+    for ((e, i) <- tokens.zipWithIndex) {
+
+      val beginOfToken = text.indexOfSlice(e, lastIndex)
+      val endOfToken = (beginOfToken + e.length) - 1
+
+      val fullPOSAnnotatorStruct = new Annotation(
+        annotatorType = annotatorType,
+        begin=beginOfToken,
+        end=endOfToken,
+        result=tags(i),
+        metadata=Map("word" -> e)
+      )
+      tokenTagAnnotation += fullPOSAnnotatorStruct
+      lastIndex = text.indexOfSlice(e, lastIndex)
+    }
+    tokenTagAnnotation
+  }
+
+  private def extractTokensAndTags: UserDefinedFunction = udf { (tokensTags: Seq[String], delimiter: String, condition: String) =>
+
+    val tempArray: ArrayBuffer[String] = ArrayBuffer()
+
+    for (e <- tokensTags.zipWithIndex) {
+      val splittedTokenTag: Array[String] = e._1.split(delimiter.mkString)
+      if(splittedTokenTag.length > 1){
+        condition.mkString match {
+          case "token" =>
+            tempArray += splittedTokenTag(0)
+
+          case "tag" =>
+            tempArray += splittedTokenTag(1)
+        }
+      }
+    }
+    tempArray
+  }
+
+  def readDataset(path: String, delimiter: String, outPutPosColName: String = "tags", outPutDocColName: String = "text"): DataFrame = {
+    import spark.implicits._
+    def annotatorType: String = AnnotatorType.POS
+
+    val tempDataFrame = spark.read.text(path).toDF
+      .filter(row => !(row.mkString("").isEmpty && row.length>0))
+      .withColumn("token_tags", split($"value", " "))
+      .select("token_tags")
+      .withColumn("tokens", extractTokensAndTags($"token_tags", lit(delimiter), lit("token")))
+      .withColumn("tags", extractTokensAndTags($"token_tags", lit(delimiter), lit("tag")))
+      .withColumn(outPutDocColName,  concat_ws(" ", $"tokens"))
+      .withColumn(outPutPosColName, annotateTokensTags($"tokens", $"tags", col(outPutDocColName)))
+      .drop("tokens", "token_tags")
+
+    tempDataFrame.withColumn(
+      outPutPosColName,
+      wrapColumnMetadata(tempDataFrame(outPutPosColName), annotatorType, outPutPosColName)
+    )
+  }
+
+  // For testing purposes when there is an array of tokens and an array of labels
+  def readDatframe(posDataframe: DataFrame, tokensCol: String = "tokens", labelsCol: String = "labels",
```

Comment:
I create this function for our internal testing where we have an array of tokens and an array of labels instead of a path to text files.

Reply:
Wondering how to make the import of this functions more intuitive. Maybe `import com.johnsnowlabs.nlp.training` ?